### PR TITLE
Database Migration and Model Update for Organization Webhook 

### PR DESCRIPTION
**Pull request by cyhsutw**
Hello guys,

This pull request is a part of #667. 

Summary of changes:

- Create two webhook related fields for `Organization`
- Add two methods to `create` and `delete` webhook on GitHub
- Add related test cases

Would love to hear your feedback on this 😄

### CODE REVIEW

File: app/models/organization.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@ class Organization < ActiveRecord::Base
 
   validates :slug, uniqueness: true
 
+  validates :webhook_id, uniqueness: true, allow_nil: true
```
Comment (cyhsutw) [inline app/models/organization.rb]: I made this optional since this feature will be placed under Flipper.

**Comment by cyhsutw** [comment]
Hi @tarebyte @johndbritton 

I think this pull request is ready for review. Would you mind take a look at it when you're available?

File: app/models/github_organization.rb [hunk 2]
```diff
@@ -98,6 +98,24 @@ def team_invitations_url
     "https://github.com/orgs/#{login}/invitations/new"
   end
 
+  # Public
+  #
```
Comment (tarebyte) [inline app/models/github_organization.rb]: Let's ✂️ the comments out for now, they more of place holders for me.

**Comment by tarebyte** [comment]
@cyhsutw I'll give this a proper 👀 tomorrow.

File: app/models/github_organization.rb (hunk 2 above)
Comment (cyhsutw) [inline app/models/github_organization.rb]: ✂️  in [d5902e2](https://github.com/education/classroom/pull/668/commits/d5902e23672fb8d8ceeb40b85a57566b16c7e092)

File: app/models/github_organization.rb [hunk 3]
```diff
@@ -98,6 +98,20 @@ def team_invitations_url
     "https://github.com/orgs/#{login}/invitations/new"
   end
 
+  def create_organization_webhook(config: {}, options: {})
+    GitHub::Errors.with_error_handling do
+      hook_config = github_org_hook_default_config.merge(config)
+                                                  .tap { |hash| hash[:secret] = ENV['WEBHOOK_SECRET'] }
```
Comment (tarebyte) [inline app/models/github_organization.rb]: Let's add this to the Rails secrets.yml file so that we don't pull from the `ENV`

File: app/models/github_organization.rb [hunk 4]
```diff
@@ -112,4 +126,17 @@ def github_repo_default_options
       organization:  @id
     }
   end
+
+  def github_org_hook_default_config
+    {
```
Comment (tarebyte) [inline app/models/github_organization.rb]: Let's just move this to one line.

```ruby
def github_org_hook_default_config
  { content_type: 'json' }
end
```

File: db/migrate/20160620141513_add_webhook_to_organizations.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class AddWebhookToOrganizations < ActiveRecord::Migration
+  def change
+    add_column :organizations, :webhook_id, :string, unique: true
```
Comment (tarebyte) [inline db/migrate/20160620141513_add_webhook_to_organizations.rb]: Will this allow Orgs with blank `webhook_id` columns?

**Comment by tarebyte** [comment]
@cyhsutw this is looking ✨.

File: db/migrate/20160620141513_add_webhook_to_organizations.rb (hunk 5 above)
Comment (cyhsutw) [inline db/migrate/20160620141513_add_webhook_to_organizations.rb]: Thanks for the catch! Actually the `unique: true` has no effect since it's an invalid option for `add_column` 😓 

Reference: http://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_column

**Comment by cyhsutw** [comment]
@tarebyte Thanks for the feedback and suggestions.

Changes have been made according to your comments.

I would really appreciate it if you could have a  👀  at them when you're available.

File: app/models/organization.rb [hunk 6]
```diff
@@ -41,4 +45,14 @@ def github_organization
   def slugify
     self.slug = "#{github_id} #{title}".parameterize
   end
+
+  def create_organization_webhook(webhook_url)
+    webhook = github_organization.create_organization_webhook(config: { url: webhook_url })
+    update_attributes(webhook_id: webhook.id)
+  end
```
Comment (tarebyte) [inline app/models/organization.rb]: I don't see where this is method is called. Is will this be used in another PR?

**Comment by tarebyte** [comment]
@cyhsutw I posted one last question, and then I think this'll be good to ship!

File: app/models/organization.rb (hunk 6 above)
Comment (cyhsutw) [inline app/models/organization.rb]: Yes, this will be used in the next PR.

**Comment by tarebyte** [comment]
```
  ActiveRecord::SchemaMigration Load (13.4ms)  SELECT "schema_migrations".* FROM "schema_migrations"
Migrating to AddWebhookToOrganizations (20160620141513)
   (1.7ms)  BEGIN
== 20160620141513 AddWebhookToOrganizations: migrating ========================
-- add_column(:organizations, :webhook_id, :string)
   (8.3ms)  ALTER TABLE "organizations" ADD "webhook_id" character varying
   -> 0.0091s
-- add_column(:organizations, :is_webhook_active, :boolean, {:default=>false})
   (49.0ms)  ALTER TABLE "organizations" ADD "is_webhook_active" boolean DEFAULT 'f'
   -> 0.0561s
== 20160620141513 AddWebhookToOrganizations: migrated (0.0655s) ===============

  SQL (5.1ms)  INSERT INTO "schema_migrations" ("version") VALUES ($1)  [["version", "20160620141513"]]
   (16.5ms)  COMMIT
```

Migration was successful.

**Comment by cyhsutw** [comment]
Thanks a lot @tarebyte!
